### ISSUE TITLE
Legacy JS transpliler support

### ISSUE THREAD

**Issue by gciatto** [mpetuska/npm-publish#49]
Hi @mpetuska,
apologies for not being helpful at all.

I started using your plugin in my project, and I soon realised you've discarded the legacy JS transpiler support.

This is a no-go for me: the new IR is too experimental and it does not fully support some Kotlin features I use in my projects... hence I cannot switch to the new IR, yet.

If I take care of restoring & maintaining legacy transpiler support in this project, would that be ok, or is there any "political" reason this should be discouraged?

**Comment by mpetuska** [comment]
Hi. I may have underexplained that legacy support termination for 3.x version. I still fully intend to maintain it for 2.x versions with patches and hotfoxes. I simply dropped legacy backend from the features standpoint.

There's a cutoff branch for old 2.x versions in the repo

**Comment by mpetuska** [comment]
P.S. no need to apologise for being unable to contribute, all contributions are just a bonus, not an expectation.

I think if you could at least raise issues against 2.x legacy backend once you encounter I could make sure they're patched out asap. I'm not using the legacy backend at all these days, so can't really spot any issues with it 😀

**Comment by gciatto** [comment]
Ok, I see, thanks.

Next question is: what's the purpose of maintaining two lines of development for the two functionalities?

Wouldn't it be easier for both users & maintainers to always use the last plugin, assuming that it supports legacy IR too?

Is there any technical limitation I'm not aware of which prevents the co-existence of legacy IR support and new IR support within your `npm-publish` plugin?

**Comment by mpetuska** [comment]
The main reason is to simplify the 3.x codebase moving forward. Having to support both, legacy and IR backends overcomplicated lots of features, required double the testing and as such was significantly slowing me down. So to maintain a decent pace, I've decided to drop support for the legacy backend moving forward (new features) and just leave it in 2.x branch to be supported with occasional pathches when nad if people report issues with it.

So in short - legacy backend support bloats the codebase too much to be maintainable.

**Comment by gciatto** [comment]
Oh I see. Makes sense! Thanks for clarifying that.

So can I attempt to add features to the 2.x line?

**Comment by mpetuska** [comment]
If you do want to add features for legacy then yes, 2.x branch is what you need. I simply don't have the bandwith doo do that myself :smile:

**Comment by gciatto** [comment]
Know the feeling :)

Ok I guess I'll try in a year or two :)
